src/ai/defaultai.cc

```diff
@@ -2779,14 +2962,22 @@ bool DefaultAI::construct_building(const Time& gametime) {
 				prio += management_data.neuron_pool[44].get_result_safe(bf->military_score_ / 20) / 5;
 
 				// Some productionsites strictly require supporting sites nearby
+				uint8_t number_of_supporters_nearby = 0;
 				if (bo.requires_supporters) {
-					uint16_t supporters_nearby = 0;
-					for (auto output : bo.ware_outputs) {
-						supporters_nearby += bf->supporters_nearby.at(output);
-					}
-					if (supporters_nearby == 0) {
+					if (bf->supporters_nearby.count(bo.name) == 0) {
 						continue;
 					}
+					number_of_supporters_nearby += bf->supporters_nearby.at(bo.desc->name());
+				}
+
+				// Priorities will be adjusted according to nearby buildings needing support
+				uint8_t number_of_supported_producers_nearby = 0;
```

Comment:
As this is replacing the former detection of the vicinity to buildings of the own type I believe this algorithm or the usage of this variable is not correct.